portfolio/src/main/java/com/google/sps/servlets/DeleteData.java

```diff
@@ -0,0 +1,39 @@
+package com.google.sps.servlets;
+
+import java.io.IOException;
+
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+
+/**
+ * Delete data from datastore object when called. 
+ */
+@WebServlet("/deletedata")
+public class DeleteData extends HttpServlet {
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response)
+      throws IOException {
+    DatastoreService storeData = DatastoreServiceFactory.getDatastoreService();
+    // TODO(morleyd): See if this code generalizes for >500 entries, think there
+    // might be an upper limit for number of entries deleted at a time
+
+    // TODO(morleyd): See if there is a better way to mass delete values, this
+    // seems like a common task and there is likely some optimization to do it
+    // faster
+    Query toDelete = new Query("Comment");
+    PreparedQuery results = storeData.prepare(toDelete);
+    for (Entity entity : results.asIterable()) {
+      storeData.delete(entity.getKey());
```

Comment:
Note that since you're retrieving entries and then separately deleting those entries, you could run into problems if you had multiple servlets accessing the same datastore. One servlet might delete the data in between the read and delete actions of another servlet. This is fine as a starting point since you're only running one servlet, but it's always something to think about when interacting with a database.

Reply:
Good point. So some sort of lock would be the way to go to then, should I look into locking mechanisms at the database level or do they just need to be at the java code level (still potential for race in this case if there's another way to access the database ie the admin console)?

Reply:
Or just look for a one shot method that's thread safe would prob be the way to

Reply:
Transactions are what you want:
https://cloud.google.com/appengine/docs/standard/java/datastore/transactions
https://cloud.google.com/appengine/docs/standard/java/javadoc/com/google/appengine/api/datastore/Transaction.html

But, feel free to save this improvement for another PR.

Reply:
I think this may be more current documentation: https://cloud.google.com/datastore/docs/concepts/transactions